Update description for deny_null_bind on ldap auth backend config (#1043)
hashicorp/web-unified-docs: content/vault/v1.10.x/content/docs/auth/ldap.mdx

- `userdn` (string, optional) - Base DN under which to perform user search. Example: `ou=Users,dc=example,dc=com`
- `userattr` (string, optional) - Attribute on user attribute object matching the username passed when authenticating. Examples: `sAMAccountName`, `cn`, `uid`
- `userfilter` (string, optional) - Go template used to construct a ldap user search filter. The template can access the following context variables: \[`UserAttr`, `Username`\]. The default userfilter is `({{.UserAttr}}={{.Username}})` or `(userPrincipalName={{.Username}}@UPNDomain)` if the `upndomain` parameter is set. The user search filter can be used to restrict what user can attempt to log in. For example, to limit login to users that are not contractors, you could write `(&(objectClass=user)({{.UserAttr}}={{.Username}})(!(employeeType=Contractor)))`.

> - `deny_null_bind` (bool, optional) - By default, Vault will prevent LDAP authentication attempts where the user provides an empty password (null binds). Setting this parameter to false allows Vault to support LDAP anonymous bind operations, which may be required for certain directory configurations that use anonymous search or discovery. When set to false, Vault defers the handling of empty-password authentication attempts to the LDAP server. The default is `true`.

Reviewer: ```suggestion
- `deny_null_bind` `(bool: true)` – By default, Vault prevents LDAP authentication 
  attempts when the user provides an empty password (null binds). Setting
  `deny_null_bind` to `false` tells Vault to defer the handling of empty-password
  authentication attempts to the LDAP server. You may want to allow LDAP
  anonymous bind operations for directory configurations using anonymous search
  or discovery.
```
Style correction: write in active voice
